app/code/Magento/InventoryInStorePickup/Model/Carrier/Command/GetIsAnyPickupLocationAvailable.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\Carrier\Command;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryInStorePickup\Model\GetPickupLocations;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Check if any pickup location is available in current scope.
+ */
+class GetIsAnyPickupLocationAvailable
+{
+    /**
+     * @var GetPickupLocations
+     */
+    private $getPickupLocations;
+
+    /**
+     * @var StockResolverInterface
+     */
+    private $stockResolver;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * Runtime cached values.
+     *
+     * @var array
+     */
+    private $isAvailable = [];
```

Comment:
Runtime caching.
Seems strange that command has state but added due to multiple execution of service.